core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1576,7 +1576,10 @@ class UnifiedLog(@volatile var logStartOffset: Long,
     val startMs = time.milliseconds
 
     def shouldDelete(segment: LogSegment, nextSegmentOpt: Option[LogSegment]): Boolean = {
-      startMs - segment.largestTimestamp > retentionMs
+      val largestTimestamp = segment.largestTimestamp
+      val shouldDelete = startMs - largestTimestamp > retentionMs
+      debug(s"$segment retentionMs breached: $shouldDelete, startMs=$startMs, largestTimestamp=$largestTimestamp, retentionMs=$retentionMs")
```

Comment:
`LogSegment#toString` already include `lastModifiedTime` and `largestRecordTimestamp`. Maybe we can delete `largestTimestamp` as it is redundant ?

Reply:
You are right, thanks.